format/format.go

```diff
@@ -75,24 +75,44 @@ func File(fset *token.FileSet, file *ast.File, goVersion string) {
 	}
 	pre := func(c *astutil.Cursor) bool {
 		f.applyPre(c)
-		if _, ok := c.Node().(*ast.BlockStmt); ok {
+		switch node := c.Node().(type) {
+		case *ast.FieldList:
+			ft, _ := c.Parent().(*ast.FuncType)
+			if ft != nil && ft.Params == node {
+				// TODO: investigate if this is worthwhile
+				// f.longLineExtra = 20
+			}
+			if ft != nil && ft.Results == node {
+				f.longLineExtra = -1
+			}
+		case *ast.BlockStmt:
 			f.blockLevel++
 		}
 		return true
 	}
 	post := func(c *astutil.Cursor) bool {
-		if _, ok := c.Node().(*ast.BlockStmt); ok {
+		f.applyPost(c)
+		switch c.Node().(type) {
+		case *ast.FieldList:
+			f.longLineExtra = 0
+		case *ast.BlockStmt:
 			f.blockLevel--
 		}
 		return true
 	}
 	astutil.Apply(file, pre, post)
 }
 
-// Multiline nodes which could fit on a single line under this many
-// bytes may be collapsed onto a single line.
+// Multiline nodes which could easily fit on a single line under this many bytes
+// may be collapsed onto a single line.
 const shortLineLimit = 60
 
+// Single-line nodes which take over this many bytes, and could easily be split
+// into two lines of at least its 40%, may be split.
+const longLineBase = 100
```

Comment:
it would be better if User could be able to define how long the line.

Reply:
See https://github.com/mvdan/gofumpt/issues/2#issuecomment-522266467 - we can't and shouldn't add options.

Reply:
Consider `80`.

> So first, something to know about me: I am an eighty column purist. For me, this has nothing to do with punchcards or whatnot, but rather with type readability, which tends to result in 50-100 characters per line — and generally about 70 or so. (I would redirect rebuttals to your bookshelf, where most any line of most any page of most any book will show this to be more or less correct.) So I personally embrace the “hard 80″, and have found that the rework that that can sometimes require results in more readable, better factored code.

http://dtrace.org/blogs/bmc/2020/10/11/rust-after-the-honeymoon/

I picked some random text books from my bookshelf and found characters per line of 83, 83, 86, 78, 82.

Reply:
### Prettier

Prettier uses 80: https://prettier.io/docs/en/options.html#print-width

Prettier is only partly relevant here, since:
1) Golang style should probably not be directly influenced by JavaScript/TypeScript style.
2) Prettier sometimes makes lines longer than 80 (even though it generally does keep lines below that).

With that said, JavaScript/TypeScript is the 2nd most popular language in the world, and Prettier is (probably?) the most popular code formatter in the world.

### 2 Files Side by Side

I think that one of the main practical reasons that people prefer lines of 80 characters is that you can open two files side by side on the same monitor and have all of the code fully visible on both windows. This is useful because you can directly compare two files side by side without having to do any horizontal scrolling. Historically, I know that people have used this setup with two VIM terminals, but it is also relevant with e.g. VSCode.

Excited to see this fork merged.

Reply:
Maybe we'll go with 90 or even 80 if 100 works well and people want it to be even stricter. I definitely think that 80 would be too short to start with, especially given how Go has historically had no convention about line lengths.

Reply:
80 is way short in "this day and age" imo. 90-120 is far nicer in my experience. Generally I go for 120 but that might be a bit long for side-by-side.

Reply:
I use 100. For those arguing on what print media uses, keep in mind that code doesn't look the same as written language because you have indentation. Sure, my lines are 100 characters long but the first 4, 8, 16, etc columns are just tabs/spaces!

Reply:
FWIW I also prefer 100